adds support to supply writeOnly flag to tapeworm to optimize `save()`

tbruun opened the pull request.

tbruun committed:
adds support to supply writeOnly flag to tapeworm to optimize `save()`

tbruun committed:
change from options to counting arguments on concurrencyStrategy

spralle commented on lib/repository.js:
```diff
@@ -112,7 +114,12 @@ Repository.prototype.save = function (aggregate, commitId, callback) {
           var nextStreamVersion = stream._version + numberOfEvents;
           if (nextStreamVersion > aggregate._version) {
             // if so ask concurrency strategy if still ok or if it needs to throw
```
should only get all events here I think

tbruun committed:
change to always call openstream with writeOnly on save.

In the event checkConcurrencyStrategy is provided, and have two arguments, a new stream will be opened with all events if ConcurrencyStrategy needs to be checked.

spralle commented on lib/repository.js:
```diff
@@ -96,49 +96,70 @@ Repository.prototype.findEventsById = function (id, callback) {
+          shouldThrow = self._concurrencyStrategy(uncommittedEvents);
+          resolve(shouldThrow);
+        } else {
+          self._partition.openStream(aggregate.id, false).then(function (stream) {
```
gillar inte att vi override:ar namnet "stream"

tbruun committed:
change variable name for new stream
